components/patient-data/api/src/main/java/org/phenotips/data/PatientRepository.java

```diff
@@ -74,4 +76,9 @@
      * @return the created patient record
      */
     Patient createNewPatient(DocumentReference creator);
+
+    /**
+     * @return a collection of all the patients.
```

Comment:
On a large instance, loading all the patients is likely to take a very long time, and consume plenty of memory. There are two optimizations that can be done:

1. Instead of returning a collection, return an iterator that will only load one patient at a time. Internally, this can  just keep a list of IDs.
2. The implementation should return a lazy collection that doesn't load all the needed data at once, only when it is actually requested.